tonic/src/transport/server/incoming.rs

```diff
@@ -91,7 +91,12 @@ fn handle_accept_error(e: impl Into<crate::Error>) -> ControlFlow<crate::Error>
     let e = e.into();
     tracing::debug!(error = %e, "accept loop error");
     if let Some(e) = e.downcast_ref::<io::Error>() {
-        if e.kind() == io::ErrorKind::ConnectionAborted {
+        if matches!(
+            e.kind(),
+            io::ErrorKind::ConnectionAborted
+                | io::ErrorKind::Interrupted
+                | io::ErrorKind::WouldBlock
```

Comment:
```suggestion
                | io::ErrorKind::WouldBlock
                | io::ErrorKind::InvalidData
```
When we got a TLS handshake error, A `InvalidData` error will happen here. see: 1897